message_types/status.proto

```diff
@@ -1,128 +1,162 @@
 syntax = "proto3";
 
 message Status {
+    message Site {
+        message InverterStatus {
+            string enabled = 1;
+            string connected = 2;
+        }
+
+        float productionWatt = 1;
+        float sizeLimitWatt = 2;
+        InverterStatus inverterStatus = 3;
+    }
+    
+    message OptimizersStatus {
+        int32 total = 1;
+        int32 online = 2;
+    }
+
+    enum ManagerStatus {
+        SHUTTING_DOWN = 0;
+        ERROR = 1;
+        STANDBY = 2;
+        PAIRING = 3;
+        POWER_PRODUCTION = 4;
+        AC_CHARGING = 5;
+        NOT_PAIRED = 6;
+        NIGHT_MODE = 7;
+        GRID_MONITORING = 8;
+        IDLE = 9;
+    }
 
-  message Site {
-    message InverterStatus {
-      string enabled = 1;
-      string connected = 2;
+    message Inverters {
+        message Unit {
+            string dspSn = 1;
+            float voltage = 2;
+            OptimizersStatus optimizersStatus = 3;
+            Temperature temperature = 4;
+            int32 fan = 5;
+            string errorCode = 6;
+            int32 subSystem = 7;
+            string badPosition = 8;
+        }
+
+        message Temperature {
+            message Units {
+                bool celsius = 1;
+                bool farenheit = 2;
+            }
+            
+            int32 value = 1;
+            Units units = 2;
+        }
+
+        Unit left = 1;
+        Unit primary = 2;
+        Unit right = 3;
     }
 
-    float productionWatt = 1;
-    float sizeLimitWatt = 2;
-    InverterStatus inverterStatus = 3;
-  }
-
-  message OptimizersStatus {
-    int32 total = 1;
-    int32 online = 2;
-  }
-
-  message Inverters {
-    message Unit {
-      string dspSn = 1;
-      float voltage = 2;
-      OptimizersStatus optimizersStatus = 3;
-      Temperature temperature = 4;
-      int32 fan = 5;
-      string errorCode = 6;
-      int32 subSystem = 7;
-      string badPosition = 8;
+    message Communication {
+        message Rs4851 {
+            message Connected {
+
+            }
+            message Enabled {
+
+            }
+            message Protocol {
+                bool seSlave = 1;
+                bool seMaster = 2;
+                bool modbusMultiDevices = 3;
+                bool sunspec = 4;
+                bool none = 5;
+            }
+            Protocol protocol = 3;
+        }
+
+        message Rs4852 {
+            message Connected {
+
+            }
+            message Enabled {
+
+            }
+            message Protocol {
+                bool seSlave = 1;
+                bool seMaster = 2;
+                bool modbusMultiDevices = 3;
+                bool sunspec = 4;
+                bool none = 5;
+            }
+            Protocol protocol = 3;
+        }
+        
+        //Todo: add more information about modbus and slaves
+        Rs4851 rs4851 = 1;
+        Rs4852 rs4852 = 2;
+        int32 lan = 3;
```

Comment:
```
communication {
  rs4851 {
    protocol {
      seSlave: true
    }
  }
  rs4852 {
    protocol {
      seSlave: true
    }
  }
  5 {
    1: "my-wifi-ssid"
    2: 4
  }
}
```

```protobuf
    message WiFi {
        string ssid = 1;
        // int32 ??
    }

    ...
    WiFi wifi = 5;
```

Reply:
5.2 is a signal strength indicator. 4 seems to be 100%/full bars.

Reply:
Added. Thank you for reporting it.